[Bug] Normalize Hunting Index Link Generation

terrancedejesus opened the pull request:
## Related Issues
* https://github.com/elastic/ia-trade-team/issues/280

## Summary
This pull request fixes a bug in the hunting folder where reference links in the `index.md` file were generating according to the platform `generate_markdown.py` was executed on.

courtesy of @Aegrah 
![image (3)](https://github.com/elastic/detection-rules/assets/99630311/ddd63d2c-c4f4-4ce7-b688-0c161a1f41c4)

## Additional Information
- Added `Source File` to metadata. Originally these were only generated in the LLM docs because they had references. Rather, we should do it globally across rules so visitors can easily from from MD to TOML since URL links are rendered. The original bug existed here as well since it was a generated link. Had to regenerate all docs as a result.


## Testing
1. Checkout branch
2. Run `python hunting/generate_markdown.py` on a Windows platform

Aegrah commented:
Thank you @terrancedejesus, will let the actual developers do the review, however.

Mikaayenson commented on hunting/generate_markdown.py:
```diff
@@ -29,12 +29,12 @@ class Hunt:
     author: str
     description: str
-    integration: list[str]
+    integration: List[str]
```
In py3.12 you don't need to use any Typing imports. 'list' is correct.

Mikaayenson reviewed (approved):
Small typo in the summary and you don't need to change the type hints. Otherwise Lgtm.

Aegrah commented:
@terrancedejesus some weird stuff is going on somewhere. This one new rule I added probably just got merged in too late (I think):
<img width="697" alt="image" src="https://github.com/elastic/detection-rules/assets/78494512/67db6180-6f7b-4a68-97ec-8e42303ad0e1">

But this seems to be an issue:
<img width="753" alt="image" src="https://github.com/elastic/detection-rules/assets/78494512/3b3af3df-fd75-4e0d-93ca-36fc12ea881c">

And that is the case for several:

<img width="1491" alt="image" src="https://github.com/elastic/detection-rules/assets/78494512/5b5a9a4a-19ab-4d47-8fd2-2f16bc3e6a32">

Aegrah reviewed (changes requested):
@terrancedejesus some weird stuff is going on somewhere. This one new rule I added probably just got merged in too late (I think):
<img width="697" alt="image" src="https://github.com/elastic/detection-rules/assets/78494512/67db6180-6f7b-4a68-97ec-8e42303ad0e1">

But this seems to be an issue:
<img width="753" alt="image" src="https://github.com/elastic/detection-rules/assets/78494512/3b3af3df-fd75-4e0d-93ca-36fc12ea881c">

And that is the case for several:

<img width="1491" alt="image" src="https://github.com/elastic/detection-rules/assets/78494512/5b5a9a4a-19ab-4d47-8fd2-2f16bc3e6a32">

Ref:
<img width="604" alt="image" src="https://github.com/elastic/detection-rules/assets/78494512/55f161a1-4fc6-4671-b870-6360b5300340">

Seems like that issue was introduced earlier, and the bug is in the query in main as well:

<img width="721" alt="image" src="https://github.com/elastic/detection-rules/assets/78494512/450984a0-9fa4-475b-a467-e8289174cc37">

<img width="681" alt="image" src="https://github.com/elastic/detection-rules/assets/78494512/e40078f1-81de-4540-999d-1f6ac3ff96c7">

terrancedejesus commented:
@Aegrah thanks for spotting this! I removed the accented characters, which only seemed to be present where the `[a-zA-Z0-9]` regex pattern in 5 separate files, but throughout multiple queries in those files. It was `[a-zA-Z0-9ñ` originally and then when loaded and MD docs generated again, changed to the accented `A` and `+`. I did not identify when those were added, but they have been fixed. Can you confirm that everything is good on your end now?

Aegrah commented:
@terrancedejesus, ran the markdown generator and now everything looks g2g. Nice.

Aegrah reviewed (approved):
Might be wise to wait for another code review, but from my testing it LGTM.
